projects/plugins/jetpack/modules/subscriptions/subscribe-modal/subscribe-modal.php

```diff
@@ -0,0 +1,196 @@
+<?php
+/**
+ * Subscribe modal popup feature
+ *
+ * @package automattic/jetpack-mu-wpcom
+ * @since 2.2.1
+ */
+
+/**
+ * Adds a Subscribe Modal template part that can be edited in the editor.
+ * This is later loaded in a pop up modal for Newsletter sites.
+ *
+ * @return void
+ */
+function wpcom_create_subscribe_template() {
```

Comment:
Do we need any function checks and bail out, or everything we need is supported in WP versions that Jetpack supports? Assume Gutenberg plugin gets disabled.

Reply:
Good question. I have added a check for wp_is_block_theme() based on separate feedback below. As is, we're now checking if it's a block theme, if the modal option is turned on, and if (the site intent is newsletter or the theme is lettre). Off the top, I'd think that would be enough checks, though I'm happy to add others if we think doing so is warranted.